### PR TITLE
SONARPY-567 Rule S5706: Special method "__exit__" should not re-raise…

### PR DESCRIPTION
**Pull request by gyula-sallai-sonarsource**
… the provided exception

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/NoReRaiseInExitCheck.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    Name name = parameter.name();
+    if (name != null) {
+      return name.symbol();
+    }
+
+    return null;
+  }
+
+  private static Symbol extractCaughtExceptionParameter(ParameterList parameterList) {
+    if (parameterList.nonTuple().size() != 4) {
```
Comment (pynicolas) [inline python-checks/src/main/java/org/sonar/python/checks/NoReRaiseInExitCheck.java]: I believe I can create a case with some Python 2 tuple parameters which make that test not quite correct. However, I think it's not worth trying to cover such corner cases.

File: its/ruling/src/test/resources/expected/python-S5706.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+'project:S5706.py':[
```
Comment (pynicolas) [inline its/ruling/src/test/resources/expected/python-S5706.json]: We usually don't do that anymore. I guess it was a good exercise in this case :)
